src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -150,7 +150,19 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     if language is not None and language.lower() == 'dotnetcore':
         site_config.app_settings.append(NameValuePair(name='ANCM_ADDITIONAL_ERROR_PAGE_LINK',
                                                       value='https://{}.scm.azurewebsites.net/detectors'.format(name)))
-
+    # This is to keep the existing appsettings for a newly created webapp on existing webapp name.
+    name_validation = client.check_name_availability(name, 'Site')
+    if not name_validation.name_available:
+        existing_app_settings = _generic_site_operation(cmd.cli_ctx, resource_group_name,
+                                                        name, 'list_application_settings')
+        # To fetch the newly added appsetting names
```

Comment:
This seems be rather complex and non performant way to fix this issue. The problem here is this line https://github.com/Azure/azure-cli/blob/7b849de40f3958a2a69b7b5ef63a5968bb3d8bf6/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L94
   ` site_config = SiteConfig(app_settings=[])`
when ever the create_webapp us called for an existing app we always reset the app settings to [] , i would fix this by making a call to get list of appsettings (that you are already doing here) if the call returns 404 exception - then i would set this value to [] as is, if not set this to value returned from the API - & leave the rest as is.